### ISSUE TITLE
[FR] easier switching between inplace / out-of-place ops

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#26644]
## 🚀 Feature

Often times I find myself writing this pattern:
```py
if training:
  with torch.enable_grad():
    # build a graph, with out of place ops on input tensors
else:
  with torch.no_grad():
    # update inplace on input tensors
```

The context manager can be unified via `torch.set_enabled_grad(training)`, yet there is no easy way to switch between in-place and out-of-place ops. 

Maybe we could, similar to `nn.ReLU(inplace)`, adding a kwarg to ops that supports inplace variants?

**Comment by vadimkantorov** [comment]
One problem is that some ops support inplace, but cannot be chained together without additional work, e.g. ReLU+Dropout (both inplace) [is not supported](https://discuss.pytorch.org/t/relu-dropout-inplace/13467/9) without resorting to https://gist.github.com/vadimkantorov/360ece06de4fd2641fa9ed1085f76d48

**Comment by gchanan** [comment]
would it work to do something like:

```
y = torch.sigmoid(x, out=(None if training else x))
```

**Comment by SsnL** [comment]
@gchanan Oh that is smart! Thanks!
